lib/wda/webdriveragent.js

```diff
@@ -365,7 +376,16 @@ class WebDriverAgent {
     if (actualUpgradeTimestamp && upgradedAt && _.toLower(`${actualUpgradeTimestamp}`) !== _.toLower(`${upgradedAt}`)) {
       log.info('Will uninstall running WDA since it has different version in comparison to the one ' +
         `which is bundled with appium-xcuitest-driver module (${actualUpgradeTimestamp} != ${upgradedAt})`);
-      return await this.uninstall();
+
+      // The default WDA_RUNNER_BUNDLE_ID can expect it is running with WDA_BUNDLE_ID
+      if (WDA_RUNNER_BUNDLE_ID === productBundleIdentifier) {
+        // Uninstall the default WDA
+        return await this.uninstall();
+      } else {
+        return await this.uninstall(productBundleIdentifier);
```

Comment:
This will most likely not work. This is not the same as the bundle id of the runner. Have you tested this locally?